src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_CollectionExpression.cs

```diff
@@ -138,6 +141,22 @@ static BoundNode unwrapListElement(BoundCollectionExpression node, BoundNode ele
             }
         }
 
+        private bool IsSingleReadOnlySpanSpread(BoundCollectionExpression node, [NotNullWhen(true)] out BoundExpression? spreadExpression)
+        {
+            spreadExpression = null;
+
+            if (node.Elements is
+                [
+                    BoundCollectionExpressionSpreadElement { Expression: { Type: NamedTypeSymbol spreadType } expr, IteratorBody: BoundExpressionStatement { Expression: not BoundConversion { ConversionKind: not ConversionKind.Identity } } }
+                ] &&
+                spreadType.OriginalDefinition == (object)_compilation.GetWellKnownType(WellKnownType.System_ReadOnlySpan_T))
```

Comment:
I must admit, I find this pattern difficult to read with `not ... { ... not ... }` and difficult to convince myself that it is correct.

An alternative might be to check for expected values directly, something like:
```
if (node.Elements is
    [
        BoundCollectionExpressionSpreadElement { Expression: { Type: NamedTypeSymbol spreadType } expr, IteratorBody : BoundExpressionStatement { Expression: var item } }
    ] &&
    item is BoundValuePlaceholder or BoundConversion { Conversion.Kind: ConversionKind.Identity, Operand: BoundValuePlaceholder { } } &&
    spreadType.OriginalDefinition == (object)_compilation.GetWellKnownType(WellKnownType.System_ReadOnlySpan_T))
```

But that alternative is also complicated. And both approaches suffer from trying to unpack the `BoundExpression` that was created in binding.

It feels like the original approach, before commit 9, which checked the type of the spread element, was more obviously correct.

Would it make sense to simply check there is an identity conversion from the spread element expression to the builder method parameter?
```
if (node.Elements is
    [BoundCollectionExpressionSpreadElement { Expression: { Type: NamedTypeSymbol spreadType } expr }] &&
    Conversions.HasIdentityConversion(node.CollectionBuilderMethod.Parameters[0].Type, spreadType))
{
```

Reply:
I think we moved away from that check because we had prior art for checking the IteratorBody. https://github.com/dotnet/roslyn/pull/73102#discussion_r1575339795

Reply:
So what's the conclusion? Should I change anything?

Reply:
Let's simply check there is an identity conversion from the spread element expression to the builder method parameter. That check is relatively easy to read, and matches the assumption made by the caller `VisitCollectionBuilderCollectionExpression()` when this method returns `true`.
```
if (node.Elements is
    [BoundCollectionExpressionSpreadElement { Expression: { Type: NamedTypeSymbol spreadType } expr }] &&
    Conversions.HasIdentityConversion(node.CollectionBuilderMethod.Parameters[0].Type, spreadType))
{
```

Reply:
We should consider moving the other checks from `VisitCollectionBuilderCollectionExpression` to this method. (Even if we don't check those in `RewriteCollectionExpressionConversion`, we will check those later in `VisitCollectionBuilderCollectionExpression`.)
```
(!constructMethod.ReturnType.IsRefLikeType || constructMethod.Parameters[0].EffectiveScope == ScopedKind.ScopedValue)
```

Then, this method could be renamed, perhaps to `CanOptimizeSingleSpreadAsCollectionBuilderArgument`.